### ISSUE TITLE
The time cost of Tensorflow is confused

### ISSUE THREAD

**Issue by MisayaZ** [tensorflow/tensorflow#1689]
I want to improve the speed of the convolution layer. I measure the cost of two different filters.
the shape of these two filters are [1,5,3,64] and[5,5,3,64]. The computation of the latter one is five times   as much as the former one, however, the time cost are almost the same  of these two convolution layer. 
conv = tf.nn.conv2d(images, kernel1, [1, 1, 1, 1], padding='SAME') 
conv = tf.nn.conv2d(images, kernel2, [1, 1, 1, 1], padding='SAME')
what is the reason that the cost can not be improved?

**Comment by keveman** [comment]
What do you mean by "time cost"? Do you mean the time to run the operation? If so, how are you measuring it? Can you please provide some more details?

**Comment by MisayaZ** [comment]
@keveman  That is right, I mean the time to run the operation. The way which I used to measure it is just like the example of  alexnet_benchmark.py, here is the code:
`def inference(images, sep_kernels):
    conv_sepf = tf.nn.conv2d(images, sep_kernelf, [1, 1, 1, 1], padding='SAME')
   
    return conv_seps

def time_tensorflow_run(target,session):
    num_steps_burn_in = 10
    total_duration = 0.0
    for i in xrange(100 + num_steps_burn_in):
        start_time = time.time()
        _ = session.run(target)
        duration = time.time() - start_time
        if i > num_steps_burn_in:
          if not i % 10:
            print ('%s: step %d, duration = %.5f' %
               (datetime.now(), i - num_steps_burn_in, duration))
          total_duration += duration
    mn = total_duration / 100
    print ('%s:  100 steps, %.5f sec' %
         (datetime.now(),mn))

def run_benchmark():
    with tf.Graph().as_default():
        images = np.load("iamges_value.npy")
        sep_kernels = np.load("sep_value2.npy")
        conv = inference(images, sep_kernels)

        init = tf.initialize_all_variables()
        sess = tf.Session()
        sess.run(init)
         
        time_tensorflow_run(conv, sess)
    print("job done!")
        

def main(_):
    run_benchmark()


if __name__ == '__main__':
    tf.app.run()

Thanks`

**Comment by keveman** [comment]
Thanks for the details. Your methodology looks good. You are running on the CPU, yes? TensorFlow uses Eigen for CPU convolutions and cuDNN for GPU convolutions. It's quite possible that the implementation used for 1,1,3,64 convolution is not as efficient as the larger one, hence your observation. Can you try it on as GPU?

**Comment by MisayaZ** [comment]
@keveman  No , I am running on GPU gtx970. Now I also try it on CPU. The time cost of the small filter is less than the large filter on CPU.

**Comment by prb12** [comment]
Unless I'm missing something, your timing loop appears to fetch the result of the convolution each time and discard it? Depending on the image sizes you may just be measuring serialization overheads. (though I may be missing something since the function inference() in your  code snippet appears to have some typos?)

To benchmark the cost of just the convolution, do session.run(target.op)

**Comment by keveman** [comment]
@MisayaZ, it is true that the filter size [5, 5, 3, 64] entails five times more computation than [1, 5, 3, 64]. However, the actual run times depend heavily on the implementation, memory access to computation ratio etc. So there is no way to guarantee any correlation between actual computation and run times. Also, we make use of external implementation such as cuDNN for the kernels. Hopefully those libraries will keep getting better. I am closing this issue as we don't plan to do anything special in TensorFlow for this.

**Comment by MisayaZ** [comment]
@prb12 sorry, I have fix the function inference().
